app/scripts/controllers/metametrics.ts

```diff
@@ -409,11 +418,30 @@ export default class MetaMetricsController {
     const { fragments } = this.store.getState();
 
     const id = options.uniqueIdentifier ?? uuidv4();
-    const fragment = {
+    let fragment = {
       id,
       ...options,
       lastUpdated: Date.now(),
     };
+
+    /**
+     * HACK: "transaction-submitted-<id>" fragment hack
```

Comment:
Should we have unit tests for any of this new logic?

Reply:
Good question. There were no existing tests associated with the methods that have been changed in this PR.

I added tests to support existing methods and logic with addition to this new logic